tools/Microsoft.TemplateSearch.TemplateDiscovery/NuGet/NugetPackProvider.cs

```diff
@@ -45,11 +45,7 @@ internal NuGetPackProvider(string name, string query, DirectoryInfo packageTempB
 
             _searchUriFormat = $"{searchResources[0].Uri}?{query}&skip={{0}}&take={{1}}&prerelease={includePreviewPacks}&semVerLevel=2.0.0";
 
-            if (Directory.Exists(_packageTempPath))
-            {
-                throw new Exception($"temp storage path for NuGet packages already exists: {_packageTempPath}");
-            }
-            else
+            if (!Directory.Exists(_packageTempPath))
```

Comment:
These two changes allow us to re-use the already-downloaded nupkgs when re-running the application. This _really_ cuts down on cycle time.